### PR TITLE
Alerting: Add Insights tab

### PR DESCRIPTION
**Pull request by VikaCep**
**What is this feature?**

Divides the home screen into _Insights_ and _Getting Started_ sections and displays panels using the scenes framework under the `Insights` section. We display different sections based on the source of the data: **Grafana**,  **Mimir Alertmanager**, **Mimir-managed Rules** and **Mimir-managed Rules - Per Rule Group**.

**Why do we need this feature?**

In order to display insights panels as a landing feature.

**Who is this feature for?**

All users

**Which issue(s) does this PR fix?**:

Part of https://github.com/grafana/alerting-squad/issues/578

**Special notes for your reviewer:**

These changes are hidden under a **feature flag** until we complete the Insights page implementation. It can be enabled with `window.feature.enable('insights-page')`

image

image

image
image

image

--------------

Proposal from Deyan
image

### CODE REVIEW

File: public/app/features/alerting/unified/Insights.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import { EmbeddedScene, PanelBuilders, SceneFlexItem, SceneFlexLayout, SceneQueryRunner } from '@grafana/scenes';
+import { useStyles2 } from '@grafana/ui';
+
+function getScene() {
+  const queryRunner1 = new SceneQueryRunner({
+    datasource: {
+      type: 'loki',
+      //since all cloud instances have a provisioned alert state history data source,
+      //we should be able to use its uid here
+      uid: 'CKA1_2hVz',
```
Comment (VikaCep) [inline public/app/features/alerting/unified/Insights.tsx:14]: Do you think this is a good idea? do we have a unique uid for this data source or how can we obtain it?

File: public/app/features/alerting/unified/Insights.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    },
+    queries: [
+      {
+        refId: 'A',
+        expr: '{from="state-history"} | json | current="Alerting"',
```
Comment (VikaCep) [inline public/app/features/alerting/unified/Insights.tsx:19]: This query should be rewritten to show meaningful information

File: public/app/features/alerting/unified/Insights.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      type: 'prometheus',
+      uid: 'gdev-prometheus',
+    },
+    queries: [
+      {
+        refId: 'A',
+        expr: 'avg by (job, instance, mode) (rate(node_cpu_seconds_total[5m]))',
+      },
+    ],
+  });
```
Comment (VikaCep) [inline public/app/features/alerting/unified/Insights.tsx:35]: This is just to show how to display several panels but this data is not relevant

File: public/app/features/alerting/unified/Insights.tsx [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+      type: 'loki',
+      uid: 'CKA1_2hVz',
+    },
+    queries: [
+      {
+        refId: 'A',
+        expr: FIRING_QUERIES_LAST_WEEK,
+      },
+    ],
+    $timeRange: new SceneTimeRange({ from: 'now-1w', to: 'now' }),
```
Comment (VikaCep) [inline public/app/features/alerting/unified/Insights.tsx:79]: I would like to have a panel that combines the result of two queries that use a different time range to show trending information (for example, number of firing alerts 2 weeks ago compared to firing alerts last week). 
So far I couldn't find a way to do it as the `$timeRange` configuration is set at the `SceneQueryRunner` level, meaning every query performed in that runner uses the same time range, and each panel is rendered using a specific query runner. I'll need to investigate more if this is something that can be done, perhaps using transformations.

File: public/app/features/alerting/unified/Insights.tsx [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        }),
+        new SceneFlexItem({
+          height: 300,
+          body: PanelBuilders.stat().setTitle('Total Firing last week').setData(queryRunner3).build(),
+        }),
+        new SceneFlexItem({
+          width: '100%',
+          height: 300,
+          body: PanelBuilders.table()
+            .setTitle('Worst Offender Alerts - This Week')
```
Comment (armandgrillet) [inline public/app/features/alerting/unified/Insights.tsx:112]: Let's rename this to something less agressive, like "Alerts that fired the most this week".

**Comment by brendamuir** [comment]
Hey Vika-incident has it as a separate page in the left-hand nav; do we want to be consistent?

File: public/app/features/alerting/unified/Insights.tsx (hunk 4 above)
Comment (VikaCep) [reply public/app/features/alerting/unified/Insights.tsx]: Unfortunately it doesn't seem to be possible to combine two or more queries with different time ranges within a same panel. As a workaround, we are able to display queries with different time ranges in different panels.

File: public/app/features/alerting/unified/Insights.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+import { useStyles2 } from '@grafana/ui';
+
+import { getFiringAlertsScene } from './insights/grafana/FiringAlertsPercentage';
+import { getFiringAlertsRateScene } from './insights/grafana/FiringAlertsRate';
+import { getMostFiredInstancesScene } from './insights/grafana/MostFiredInstancesTable';
+import { getMostFiredRulesScene } from './insights/grafana/MostFiredRulesTable';
+import { useAlertmanager } from './state/AlertmanagerContext';
+import { GRAFANA_RULES_SOURCE_NAME } from './utils/datasource';
+
+//all cloud instances are guaranteed to have this datasource uid for the alert state history loki datasource
```
Comment (konrad147) [inline public/app/features/alerting/unified/Insights.tsx:15]: What about OSS self-hosted instances? We should probably get this information from the config

File: public/app/features/alerting/unified/Home.tsx [hunk 7]
```diff
@@ -1,271 +1,48 @@
...
-              <li>If the condition is breached, an alert instance fires</li>
-              <li>Firing instances are routed to notification policies based on matching labels</li>
-              <li>Notifications are sent out to the contact points specified in the notification policy</li>
-            </ul>
-          </div>
-          <SVG
-            src={`public/img/alerting/at_a_glance_${theme.name.toLowerCase()}.svg`}
-            width={undefined}
-            height={undefined}
+    <AlertmanagerPageWrapper pageId={'alerting'} accessType={'instance'}>
```
[the remark this answers is not in the document]
Comment (VikaCep) [reply public/app/features/alerting/unified/Home.tsx]: This has been removed as we decided not to show the alertmanager selector on top

File: public/app/features/alerting/unified/Insights.tsx (hunk 6 above)
Comment (VikaCep) [reply public/app/features/alerting/unified/Insights.tsx]: Agree although we decided to initially focus only on cloud customers

File: public/app/features/alerting/unified/insights/grafana/FiringAlertsPercentage.tsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            right: 'Value #B',
+          },
+          replaceFields: true,
+        },
+      },
+    ],
+  });
+
+  return new SceneFlexItem({
+    width: 'calc(50% - 8px)',
```
[the remark this answers is not in the document]
Comment (VikaCep) [reply public/app/features/alerting/unified/insights/grafana/FiringAlertsPercentage.tsx]: I didn't find a `gap` property in either `SceneFlexLayout` or `SceneFlexItem`, maybe I'm missing something 🤔

File: public/app/features/alerting/unified/Insights.tsx [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+  const styles = useStyles2(getStyles);
+  const grafanaScenes = getGrafanaScenes();
+  const cloudScenes = getCloudScenes();
+  const mimirRulesScenes = getMimirManagedRulesScenes();
+  const mimirRulesPerGroupScenes = getMimirManagedRulesPerGroupScenes();
+
+  return (
+    <>
+      <div className={styles.container}>
+        <h4>Grafana</h4>
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/Insights.tsx:116]: nit: I think we generally discourage the usage of `H(N)` in favor of the `<Text variant='h4>Hello, world!</Text>` if we can't guarantee the document will also have a H1 through H3 preceding it.

File: public/app/features/alerting/unified/insights/grafana/FiringAlertsPercentage.tsx
Comment (gillesdemey) [inline public/app/features/alerting/unified/insights/grafana/FiringAlertsPercentage.tsx:1]: ❤️  the separate files for each visualization!

File: public/app/features/alerting/unified/insights/grafana/MostFiredInstancesTable.tsx [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+      },
+    ],
+
+    $timeRange: timeRange,
+  });
+
+  const ruleLinkTransformation: CustomTransformOperator = () => (source: Observable<DataFrame[]>) => {
+    return source.pipe(
+      map((data: DataFrame[]) => {
+        return data.map((frame: DataFrame) => {
```
[the remark this answers is not in the document]
Comment (gillesdemey) [reply public/app/features/alerting/unified/insights/grafana/MostFiredInstancesTable.tsx:38]: +1 this bit is hard to parse :D I think some code comments that explain what we're trying to do here might also help

**Review by gillesdemey — APPROVED** [review]
LGTM – only a few minor nits

File: public/app/features/alerting/unified/features.ts [hunk 11]
```diff
@@ -6,6 +6,7 @@ export enum AlertingFeature {
   NotificationPoliciesV2MatchingInstances = 'notification-policies.v2.matching-instances',
   DetailsViewV2 = 'details-view.v2',
   ContactPointsV2 = 'contact-points.v2',
+  InsightsPage = 'insights-page',
```
Comment (torkelo) [inline public/app/features/alerting/unified/features.ts:9]: @VikaCep in the future please use the built-in feature toggle system Grafana has, as far as I am aware all other teams use it. With it you can create feature toggles that can be set in the ini file, via ENV variables, and via URL query parameter, and viewed in the UI. 

https://github.com/grafana/grafana/blob/main/pkg/services/featuremgmt/registry.go

File: public/app/features/alerting/unified/home/Insights.tsx [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+      </CollapsableSection>
+
+      <CollapsableSection label="Mimir-managed Rules" isOpen={true}>
+        <mimirRulesScenes.Component model={mimirRulesScenes} />
+      </CollapsableSection>
+
+      <CollapsableSection label="Mimir-managed Rules - Per Rule Group" isOpen={true}>
+        <mimirRulesPerGroupScenes.Component model={mimirRulesPerGroupScenes} />
+      </CollapsableSection>
+    </div>
```
Comment (torkelo) [inline public/app/features/alerting/unified/home/Insights.tsx:152]: I would try to break the scene dashboard up into separate tabs so you do not have to scroll so much, but just a thought

File: public/app/features/alerting/unified/features.ts (hunk 11 above)
Comment (VikaCep) [reply public/app/features/alerting/unified/features.ts:9]: Good call. Changed it to use the built-in feature toggle in https://github.com/grafana/grafana/pull/74749
